tools/radar_sync.py

```diff
@@ -0,0 +1,404 @@
+                port=self._port,
+                username=self._username,
+                key_filename=self._key_file
+            )
+
+            _folder = '{}/{}/{}/{}'.format(self._root_dir, year, month, day)
+
+            _, stdout, _ = self._client.exec_command('ls {}'.format(_folder))
+            _contents = []
+
+            for line in stdout:
+                _contents.append(line.strip('\n'))
+
+            self._client.close()
+        except Exception as ex:
+            logging.critical("*** Caught exception: %s: %s", ex.__class__, ex)
+            try:
+                self._client.close()
+            except Exception:
+                pass
+
+        return _contents
+
+    def get_folder_size(self, year='', month='', day=''):
+        """
+        Gets folder size in MB.
+        """
+        _content = 0
+
+        try:
+            self._client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            self._client.connect(
+                hostname=self._hostname,
+                port=self._port,
+                username=self._username,
+                key_filename=self._key_file
+            )
+
+            _folder = '{}/{}/{}/{}'.format(self._root_dir, year, month, day)
+
+            _, stdout, _ = self._client.exec_command(
+                'du -ms {}'.format(_folder))
+
+            for line in stdout:
+                _content, _ = line.strip('\n').split()
+
+            self._client.close()
+        except Exception as ex:
+            logging.critical("*** Caught exception: %s: %s", ex.__class__, ex)
+            try:
+                self._client.close()
+            except Exception:
+                pass
+
+        return int(_content)
+
+    def remote_to_local_copy(self, folder):
+        """
+        Copies the contents of a remote folder."
+        """
+        _transport = paramiko.Transport((self._hostname, self._port))
+        _r_key = paramiko.RSAKey.from_private_key_file(self._key_file)
+
+        try:
+            _transport.connect(
+                username=self._username,
+                pkey=_r_key
+            )
+
+            _client = paramiko.SFTPClient.from_transport(_transport)
+            _remote_path = '{}/{}'.format(self._root_dir, folder)
+            _local_path = os.path.basename(_remote_path)
+            _client.chdir(_remote_path)
+
+            if not os.path.isdir(_local_path):
+                os.mkdir(_local_path)
+
+            for _file in _client.listdir("."):
+                if stat.S_ISREG(_client.stat(_file).st_mode):
+                    logging.debug("Retrieving SFTP remote file '%s'", _file)
+                    _client.get(_file, f'{_local_path}/{_file}')
+
+        except Exception as ex:
+            logging.critical("*** Caught exception: %s: %s", ex.__class__, ex)
+            traceback.print_exc()
+            _transport.close()
+            sys.exit(1)
+
+        _transport.close()
+
+
+def check_hdfs_url(hdfs_url):
```

Comment:
This is not necessary and potentially harmful. Better let the user's path through and let Pydoop / Hadoop handle it